src/collections/_documentation/development/sdk-dev/envelopes.md

```diff
@@ -0,0 +1,448 @@
+
+```
+{"event_id":"9ec79c33ec9942ab8353589fcb2e04dc"}\n
+{"type":"attachment"}\n
+helloworld
+```
+
+**Envelope without headers, implicit length, last newline omitted, terminated by
+EOF:**
+
+```
+{}\n
+{"type":"session"}\n
+{"started": "2020-02-07T14:16:00Z","attrs":{"release":"sentry-test@1.0.0"}}
+```
+
+## Data Model
+
+Each Envelope consists of headers and a potentially empty list of items, each
+with their own headers. Which Headers are required depends on the Items in an
+Envelope. This section describes all Item types and their respective required
+headers.
+
+The type of an item is declared in the `type` header, as well as the payload
+size in `length`. See Serialization Format for a list of common Item headers.
+The headers described in this section are **in addition to the common headers**.
+
+### Event
+
+Item type `"event"`. This Item contains an error or default event payload
+encoded in JSON.
+
+**Constraints:**
+
+- This item may occur at most once per Envelope.
+- This item is mutually exclusive with `"transaction"` items.
+
+**Envelope Headers:**
+
+`event_id`
+
+: **UUID String, required.** Corresponds to the `event_id` field of the event
+  payload. Clients are required to generate an event id ahead of time and set it
+  at least in the Envelope headers. If the identifier mismatches between the
+  Envelope and payload, the Envelope header takes precedence.
+  
+`sent_at`
+
+: *String, recommended.* The timestamp when the event was sent from the SDK as
+  [RFC 3339](https://tools.ietf.org/html/rfc3339) string. Used for clock drift
+  correction of the event timestamp. Timezone must be UTC.
+
+**Additional Item Headers:**
+
+*None*
+
+### Transaction
+
+Item type `"transaction"`. This Item contains a transaction event payload
+encoded in JSON.
+
+**Constraints:**
+
+- This item may occur at most once per Envelope.
+- This item is mutually exclusive with `"event"` items.
+
+**Envelope Headers:**
+
+`event_id`
+
+: **UUID String, required.** Corresponds to the `event_id` field of the
+  transaction payload. Clients are required to generate an event id ahead of
+  time and set it at least in the Envelope headers. If the identifier mismatches
+  between the Envelope and payload, the Envelope header takes precedence.
+
+`sent_at`
+
+: *String, recommended.* The timestamp when the event was sent from the SDK as
+  [RFC 3339](https://tools.ietf.org/html/rfc3339) string. Used for clock drift
+  correction of the event timestamp. Timezone must be UTC.
+
+**Additional Item Headers:**
+
+*None*
+
+### Attachment
+
+Item type `"attachment"`. This Item contains a raw payload of an attachment
+file. It is always associated to an event or transaction.
+
+**Constraints:**
+
+- This item may occur multiple times per Envelope.
+- For **minidump** and **apple crash report** attachments, the corresponding
+  `"event"` item must be sent within the same Envelope.
+- Generic attachments can be ingested separately from their events. It is
```

Comment:
"We recommend sending them in the same Envelope, which allows for more efficient rate limiting and filtering."